### ISSUE TITLE
Test new MGI data (on snapshot)

### ISSUE THREAD

**Issue by kltm** [geneontology/go-site#1876]
A new MGI GAF with updated entity types will soon be made available. We would like to test it in the `master` or `snapshot` pipelines before finalizing it.

Tagging @ukemi @loricorbani

**Comment by loricorbani** [comment]
Seth,

Please test these new files.    Column 12 is now using the SO terms.  Please let me know if this isn't clear or if you need more information.  Once you are happy, I will tag and pass our changes over to MGI Public.

http://www.informatics.jax.org/downloads/custom/mdolan_gene_association.mgi
http://www.informatics.jax.org/downloads/custom/mdolan_gene_association_nonoctua.mgi

Mary : please add more info that you think is important
 
**Col 12 with SO terms:**
gene
gene_segment
genetic_marker
lincRNA_gene
lncRNA_gene
miRNA_gene
ncRNA_gene
protein_coding_gene
pseudogene
RNase_MRP_RNA_gene
RNase_P_RNA_gene
rRNA_gene
scRNA_gene
sense_intronic_ncRNA_gene
sense_overlap_ncRNA_gene
snoRNA_gene
snRNA_gene
SRP_RNA_gene
telomerase_RNA_gene
tRNA_gene

**Comment by kltm** [comment]
@loricorbani I'm assuming that the file we care about here is

http://www.informatics.jax.org/downloads/custom/mdolan_gene_association_nonoctua.mgi

being used at https://github.com/geneontology/go-site/blob/78a5d1389c433ddbf37b78cc63a0b512b5cf6baf/metadata/datasets/mgi.yaml#L28

I'm not sure where we'd use http://www.informatics.jax.org/downloads/custom/mdolan_gene_association.mgi ?

**Comment by kltm** [comment]
Should be tested with next full runs of `master` or `snapshot`.

**Comment by kltm** [comment]
Okay, testing is good. I've run the new MGI file through the pipeline and put it up on a test AmiGO. This AmiGO only gives the following types:
```
protein_coding_gene
gene_product
gene
gene segment
ncRNA-coding gene
pseudogene
biological region
protein
```
The proper things are in the src file, so it seems like it's something "fixing" things in ontobio.  @dustine32 I'm a little suspicious that it's a problem with https://github.com/biolink/ontobio/blob/7fe4a7aea5a016bd58025f56c7105ef940d76ed6/ontobio/model/association.py#L211-L247 . Do you know if SO is consumed by ontobio of if this is some kind of holdover, or maybe my guess is off?

**Comment by loricorbani** [comment]
The mdolan_gene_association.mgi that I sent is just a test copy.
I included Mary's name so it is clear that it is a test copy.
Sorry for the confusion.

**Comment by kltm** [comment]
@ukemi @loricorbani I would not make the change, as there seems to be an issue, described here https://github.com/geneontology/go-site/issues/1876#issuecomment-1178281855. If you need this out the door, we should talk timelines. For the moment, I'm going to try and coordinate w/ @dustine32 about what steps we should take.

**Comment by kltm** [comment]
It looks like http://www.informatics.jax.org/downloads/custom/mdolan_gene_association_nonoctua.mgi has been pulled from circulation? It might be good if we could still have that for testing.

**Comment by loricorbani** [comment]
See my earlier comments.  All *test* files are now in the "custom/noctua" folder.  
I renamed the "mdolan_" and moved into the custom/noctua folder.

mdolan_gene_association_nonnoctua.mgi  ->  custom/noctua/gene_association_nonoctua.mgi

http://www.informatics.jax.org/downloads/custom/noctua

**Comment by kltm** [comment]
Location for future testing: `http://www.informatics.jax.org/downloads/custom/noctua/gene_association_nonoctua.mgi`

**Comment by kltm** [comment]
SO terms in the file:
```
SO:0000165
SO:0000167
SO:0000170
SO:0000624
SO:0001055
SO:0001158
SO:0001668
SO:0001669
SO:0001952
```

**Comment by kltm** [comment]
@dustine32 @loricorbani I believe that below are the mappings:
```
docker run --network host -it obolibrary/odkfull:v1.3.1 /bin/bash
root@moiraine:/tools# runoak -i obolibrary:so.owl info "tRNA_gene"
```
```
SO:0002182 ! antisense_lncRNA_gene
SO:3000000 ! gene_segment
SO:0001645 ! genetic_marker
SO:0002127 ! lncRNA_gene
SO:0001265 ! miRNA_gene
SO:0001217 ! protein_coding_gene
SO:0000336 ! pseudogene
SO:0001266 ! scRNA_gene
SO:0001272 ! tRNA_gene
```

**Comment by kltm** [comment]
@dustine32 @balhoff 

Okay, so we're getting `gene segment` from minerva and do not have `gene_segment` otherwise:
```
release/products/annotations$ zgrep 'gene segment' *.gaf.gz | cut -f 1 | sort | uniq
noctua_mgi_valid.gaf.gz:MGI
release/products/annotations$ zgrep 'gene_segment' *.gaf.gz | cut -f 1 | sort | uniq
```
The new MGI file that @loricorbani is producing has `gene_segment`. ontobio is only going to allow us to have one of these and we should probably have them match. SO seems to should "gene_segment" as the label, so I think we should probably head that way unless there is some other reason?

**Comment by ukemi** [comment]
At one point I pinged SO about removing the underscores and they told me it was not possible. Seth, I'm not sure where the above is coming from. Does the MGI file have SO terms in it that don't map correctly because they do/don't have underscores? If so, what exactly does @loricorbani need to do to our file. I'm getting confused about what is on the MGI end and what is not.

**Comment by loricorbani** [comment]
this is what is in the MGI: file that I just generated:

gene
gene_segment
genetic_marker
lincRNA_gene
lncRNA_gene
miRNA_gene
ncRNA_gene
protein_coding_gene
pseudogene
RNase_MRP_RNA_gene
RNase_P_RNA_gene
rRNA_gene
scRNA_gene
sense_intronic_ncRNA_gene
sense_overlap_ncRNA_gene
snoRNA_gene
snRNA_gene
SRP_RNA_gene
telomerase_RNA_gene
tRNA_gene

**Comment by kltm** [comment]
@ukemi @loricorbani At this point: nothing. We first need to figure out what the correct canonical label should be (I believe the answer should be what SO provides as the primary label or the GAF 2.2 spec pretty much goes off the rails) and then work out what needs to change to get there. I believe that minerva is what's primarily giving a non-primary label, so this may end up a non-MGI fix.

**Comment by balhoff** [comment]
@kltm I checked the current release of go-lego and SO:3000000 has label `gene_segment` there. There is an exact synonym 'gene segment' but I don't think that should be coming into play.

**Comment by kltm** [comment]
@balhoff @dustine32 Okay, I'm noticing now that it's the `_valid` file and not the `-src` file, so maybe ontobio is doing the correction here. I guess that makes sense as it's probably using that map to get from GPAD to GAF. @dustine32 I guess that brings us back to trying https://github.com/biolink/ontobio/pull/626. Could you take a look at that, merge it if it makes sense and do a release? With that, we can try bumping the pipeline and see what we get.

**Comment by kltm** [comment]
@dustine32 Thank you for the merge and update. With those in place, I've started retesting on `master` (and `snapshot` will occur naturally over the weekend).

**Comment by loricorbani** [comment]
@kltm @ukemi Did the weekend test pass?

**Comment by dustine32** [comment]
@loricorbani I think it did pass based on the increase of SO term labels in col 12. Here is the current GO release from 2022-07-01:
```
$ curl -s http://current.geneontology.org/annotations/mgi.gaf.gz | zgrep -v -e "^\!" | cut -f12 | sort | uniq -c
 101 biological region
2119 gene
2714 gene segment
34275 gene_product
 275 ncRNA-coding gene
269359 protein
205608 protein_coding_gene
 537 pseudogene
```
And here is what we produced over the weekend:
```
$ curl -s http://skyhook.berkeleybop.org/snapshot/annotations/mgi.gaf.gz | zgrep -v -e "^\!" | cut -f12 | sort | uniq -c
   3 RNase_MRP_RNA_gene
  12 RNase_P_RNA_gene
   4 SRP_RNA_gene
 101 biological region
1496 gene
1200 gene_product
2865 gene_segment
   9 genetic_marker
  36 lincRNA_gene
27395 lncRNA_gene
3250 miRNA_gene
 275 ncRNA_gene
1053 protein
484841 protein_coding_gene
 542 pseudogene
  34 rRNA_gene
   5 scRNA_gene
  20 sense_intronic_ncRNA_gene
   4 sense_overlap_ncRNA_gene
  33 snRNA_gene
 486 snoRNA_gene
1815 tRNA_gene
  13 telomerase_RNA_gene
```

**Comment by kltm** [comment]
@loricorbani I think that we should fully test it all the way through the pipeline before proceeding. We're still waiting on a successful total run due to some hiccups at the end of last week.

**Comment by kltm** [comment]
@ukemi @loricorbani On the testing AmiGO, in the last run, we now have the following types:
```
protein_coding_gene
protein
lncRNA_gene
miRNA_gene
gene_segment
tRNA_gene
gene
gene_product
pseudogene
snoRNA_gene
ncRNA_gene
biological region
protein_complex
lincRNA_gene
rRNA_gene
snRNA_gene
sense_intronic_ncRNA_gene
telomerase_RNA_gene
RNase_P_RNA_gene
genetic_marker
scRNA_gene
SRP_RNA_gene
sense_overlap_ncRNA_gene
RNase_MRP_RNA_gene
```
If that looks right to you, we are ready to switch the files over.

**Comment by ukemi** [comment]
@mdolanme and @loricorbani  I believe this is what we want for now.

**Comment by pgaudet** [comment]
Can we arrange to avoid gene and protein_coding_gene

and use protein and gene_product instead? 

(we can revisit this later, but these seem non orthogonal)

**Comment by loricorbani** [comment]
installed on MGI Production

http://www.informatics.jax.org/downloads/reports/gene_association.mgi

**Comment by kltm** [comment]
Updated and moved to clearing. This can close as soon as any new build is done.
